plugins/woocommerce-admin/client/marketplace/components/content/content.tsx

```diff
@@ -1,34 +1,124 @@
-	switch ( selectedTab ) {
-		case 'extensions':
-			return <Extensions />;
-		case 'themes':
-			return <Themes />;
-		default:
-			return <Discover />;
-	}
-};
+type ContentComponentType =
+	| typeof Discover
+	| typeof Extensions
+	| typeof SearchResults
+	| typeof Themes;
 
 export default function Content(): JSX.Element {
 	const marketplaceContextValue = useContext( MarketplaceContext );
+
+	const [ productList, setProductList ] = useState< Product[] >( [] );
+	const { setIsLoading } = marketplaceContextValue;
+
 	const { selectedTab } = marketplaceContextValue;
+	const query = useQuery();
+
+	// Get the content for this screen
+	useEffect( () => {
+		if ( [ '', 'discover' ].includes( selectedTab ) ) {
+			return;
+		}
+
+		setIsLoading( true );
+		setProductList( [] );
+
+		const params = new URLSearchParams();
+
+		if ( query.term ) {
+			params.append( 'term', query.term );
+		}
+
+		if ( query.category ) {
+			params.append( 'category', query.category );
+		} else if ( selectedTab === 'themes' ) {
+			params.append( 'category', 'themes' );
+		} else if ( selectedTab === 'search' ) {
+			params.append( 'category', 'extensions-themes' );
+		}
+
+		const wccomSettings = getAdminSetting( 'wccomHelper', false );
+		if ( wccomSettings.storeCountry ) {
+			params.append( 'country', wccomSettings.storeCountry );
+		}
+
+		const wccomSearchEndpoint =
+			MARKETPLACE_HOST +
+			MARKETPLACE_SEARCH_API_PATH +
+			'?' +
+			params.toString();
+
+		// Fetch data from WCCOM API
+		fetch( wccomSearchEndpoint )
+			.then( ( response ) => response.json() )
+			.then( ( response ) => {
+				/**
+				 * Product card component expects a Product type.
+				 * So we build that object from the API response.
+				 */
+				const products = response.products.map(
+					( product: SearchAPIProductType ): Product => {
+						return {
+							id: product.id,
+							title: product.title,
+							image: product.image,
+							type: product.type,
+							description: product.excerpt,
+							vendorName: product.vendor_name,
+							vendorUrl: product.vendor_url,
+							icon: product.icon,
+							url: product.link,
+							// Due to backwards compatibility, raw_price is from search API, price is from featured API
+							price: product.raw_price ?? product.price,
+							averageRating: product.rating ?? 0,
+							reviewsCount: product.reviews_count ?? 0,
+						};
+					}
+				);
+				setProductList( products );
+			} )
+			.catch( () => {
+				setProductList( [] );
+			} )
+			.finally( () => {
+				setIsLoading( false );
+			} );
+	}, [ query.term, query.category, selectedTab, setIsLoading ] );
+
+	let ContentComponent: ContentComponentType | undefined;
+	if ( selectedTab === 'extensions' ) {
+		ContentComponent = Extensions;
+	} else if ( selectedTab === 'themes' ) {
+		ContentComponent = Themes;
+	} else if ( selectedTab === 'search' ) {
+		ContentComponent = SearchResults;
+	} else if ( selectedTab === 'discover' ) {
+		ContentComponent = Discover;
+	}
```

Comment:
I think I prefer the `switch` we had before. Would something like this work here: https://github.com/woocommerce/woocommerce/commit/e01eead8296da8ac92e75c9ccaf2a861f8291602. Just a nitpick and not a blocker.

I removed the `ContentComponent` and `ContentComponentType` - not sure if they do something special that I missed.

Reply:
Done [59456da](https://github.com/woocommerce/woocommerce/pull/40303/commits/59456da1f1be5edde8fa67e472a540d4d742cc79)